src/screens/ExchangeModal.tsx

```diff
@@ -684,14 +683,28 @@ export default function ExchangeModal({
       } else {
         const ethPriceInNativeCurrency =
           genericAssets[ETH_ADDRESS]?.price?.value ?? 0;
-        const tokenPriceInNativeCurrency =
+        const inputTokenPriceInNativeCurrency =
           genericAssets[inputCurrency?.address]?.price?.value ?? 0;
-        const tokensPerEth = divide(
-          tokenPriceInNativeCurrency,
+        const outputTokenPriceInNativeCurrency =
+          genericAssets[outputCurrency?.address]?.price?.value ?? 0;
+        const inputTokensPerEth = divide(
+          inputTokenPriceInNativeCurrency,
           ethPriceInNativeCurrency
         );
-        const inputTokensInEth = multiply(tokensPerEth, inputAmount!);
-        amountInUSD = multiply(priceOfEther, inputTokensInEth);
+        const outputTokensPerEth = divide(
+          outputTokenPriceInNativeCurrency,
+          ethPriceInNativeCurrency
+        );
+        const inputTokensInEth = multiply(inputTokensPerEth, inputAmount!);
+        const outputTokensInEth = multiply(outputTokensPerEth, outputAmount!);
+
+        const availableTokenPrice = inputTokensInEth ?? outputTokensInEth;
+        const maybeResultAmount = multiply(priceOfEther, availableTokenPrice);
+        // We have to use string matching here because the multiply helper will return the value as a string from the helpers
+        // If we pass a empty string value to segment it gets ignored
+        amountInUSD = ['NaN', '0'].includes(maybeResultAmount)
```

Comment:
this seems kinda funky but I get the reasoning 👌

Reply:
Yeah its not ideal, ideally the functions would throw or return undefined so we could check more appropriately